register/sha256/sha256.go

```diff
@@ -0,0 +1,21 @@
+// This package has no purpose except to perform registration of multihashes.
+//
+// It is meant to be used as a side-effecting import, e.g.
+//
+//	import (
+//		_ "github.com/multiformats/go-multihash/register/sha256"
+//	)
+//
+// This package an implementation of sha256 using the go std, this is recomanded
+// if you are using go1.21 or above.
+package sha256
+
+import (
+	"crypto/sha256"
+
+	multihash "github.com/multiformats/go-multihash/core"
+)
+
+func init() {
+	multihash.Register(multihash.SHA2_256, sha256.New)
```

Comment:
This is now registering twice for Go 1.20, so it depends on the order that the `init` functions are called. This is kind of subtle.

Can you just make sure `Register` is only called once (in both the Go 1.20 and 1.21 code path)?

Reply:
You would need to import `_ "github.com/multiformats/go-multihash/register/sha256"` directly in your app for this to happen, the defaults all go through `github.com/multiformats/go-multihash/register/miniosha256` for now since it has the build tags. (I will submit a closing up patch which drops this in one year once go1.22 releases.)
Since no one does this already https://pkg.go.dev/github.com/multiformats/go-multihash/register/miniosha256?tab=importedby I don't think it's a problem.
Also even if you were to do this AFAIT the second register override the first one, which is not that bad.

Also the go spec weakly defines what should happen in this case (this happens after dependency analysis):
> To ensure reproducible initialization behavior, build systems are encouraged to present multiple files belonging to the same package in lexical file name order to a compiler. 

https://go.dev/ref/spec#Package_initialization

If you really want I'll make that happen but I don't think it's an issue and worth maintaining one more go file.